index.ts

```diff
@@ -10,19 +11,29 @@ enum APIVersion {
 
 export class Targetprocess {
     private subdomain: string;
+    private accessToken: string;
     private headers: {
         [index: string]: string;
     };
 
-    constructor(subdomain: string, username: string, password: string) {
+    constructor(
+        subdomain: string,
+        username?: string,
+        password?: string,
+        config?: { accessToken?: string }
+    ) {
         this.subdomain = subdomain;
+        this.accessToken = config && config.accessToken;
```

Comment:
Please use Elvis operator.
```suggestion
        this.accessToken = config?.accessToken;
```